packages/xdl/src/TurtleApi.ts

```diff
@@ -0,0 +1,200 @@
+import QueryString from 'querystring';
+
+import FormData from './tools/FormData';
+import { JSONObject, JSONValue } from '@expo/json-file';
+import axios, { AxiosRequestConfig } from 'axios';
+import concat from 'concat-stream';
+import ExtendableError from 'es6-error';
+import fs from 'fs-extra';
+import idx from 'idx';
+import merge from 'lodash/merge';
+
+import Config from './Config';
+
+
+const apiBaseUrl = `${Config.turtleApi.scheme}://${Config.turtleApi.host}:${Config.turtleApi.port}`;
+
+export class TurtleApiError extends ExtendableError {
+  code: string;
+  details?: JSONValue;
+  serverStack?: string;
+  readonly _isApiError = true;
+
+  constructor(message: string, code: string = 'UNKNOWN') {
+    super(message);
+    this.code = code;
+  }
+}
+
+type RequestOptions = {
+  httpMethod: 'get' | 'post' | 'put' | 'delete';
+  queryParameters?: QueryParameters;
+  body?: JSONObject;
+};
+
+type QueryParameters = { [key: string]: string | number | boolean | null | undefined };
+
+type TurtleApiClientOptions = {
+  sessionSecret?: string;
+};
+
+export default class TurtleApiClient {
+  sessionSecret: string | null = null;
+
+  static clientForUser(user?: TurtleApiClientOptions | null): TurtleApiClient {
+    if (user && user.sessionSecret) {
+      return new TurtleApiClient({ sessionSecret: user.sessionSecret });
+    }
+    return new TurtleApiClient(
+    );
+  }
+
+  constructor(options: TurtleApiClientOptions = {}) {
```

Comment:
We don't provide any non authenticated endpoints so secret should be required when creating Api object